src/main/php/lang/Process.class.php

```diff
@@ -64,10 +58,36 @@ public function __construct($command= null, $arguments= [], $cwd= null, $env= nu
     $cmd= CommandLine::forName(PHP_OS);
     foreach ($cmd->resolve($command) as $binary) {
       $binary= realpath($binary);
-
-      // Resolved binary, try creating a process from it
       $exec= $cmd->compose($binary, $arguments);
-      if (!is_resource($this->handle= proc_open($exec, $spec, $pipes, $cwd, $env, ['bypass_shell' => true]))) {
+      $options= ['bypass_shell' => true];
+
+      // Default descriptor spec to map STDIN to a pipe to read from and STDOUT and STDERR to
+      // pipes the process will write to. This can be overwritten by the descriptors argument.
+      //
+      // Rewrite ['redirect', n] and ['null'] arguments for PHP versions <= 7.4.0, see
+      // https://github.com/php/php-src/commit/6285bb52faf407b07e71497723d13a1b08821352
+      $spec= [['pipe', 'r'], ['pipe', 'w'], ['pipe', 'w']];
+      foreach ((array)$descriptors as $n => $descriptor) {
+        if (PHP_VERSION_ID >= 70400 || !is_array($descriptor)) {
+          $spec[$n]= $descriptor;
+        } else if ('redirect' === $descriptor[0]) {
+          $exec.= ' '.$n.'>&'.$descriptor[1];
+          $options['bypass_shell']= false;
+        } else if ('null' === $descriptor[0]) {
+          $spec[$n]= ['file', CommandLine::$WINDOWS === $cmd ? 'NUL' : '/dev/null', 'w'];
+        } else {
+          $spec[$n]= $descriptor;
+        }
+      }
+
+      // For non-Windows systems, use `exec` to replace the extra /bin/sh between this and the
+      // executed process, see https://www.php.net/manual/de/function.proc-get-status.php#93382
+      if (CommandLine::$WINDOWS !== $cmd && $options['bypass_shell']) {
+        $exec= 'exec '.$exec;
```

Comment:
Before:
```bash
$ xp -w '(new lang\Process("pstree", [getmypid()], null, null, [STDIN, STDOUT, STDERR]))->close()'
php───sh───pstree
```

After:
```bash
$ xp -w '(new lang\Process("pstree", [getmypid()], null, null, [STDIN, STDOUT, STDERR]))->close(
)'
php───pstree
0
```

Reply:
Verified in xp-framework/core@ea14f37f5, see https://github.com/xp-framework/core/actions/runs/1274909999